lms/views/api/canvas_studio.py

```diff
@@ -0,0 +1,118 @@
+"""
+Views for authorizing with Canvas Studio and listing videos.
+
+See `CanvasStudioService` for more details.
+"""
+
+from pyramid.httpexceptions import HTTPFound
+from pyramid.view import view_config
+
+from lms.security import Permissions
+from lms.services import CanvasStudioService
+from lms.validation.authentication import OAuthCallbackSchema
+
+
+# View for authorization popup which redirects to Canvas Studio's OAuth
+# authorization endpoint.
+#
+# Canvas Studio's OAuth implementation has several issues which complicate
+# using the standard OAuth flow:
+#
+# 1. Canvas Studio does not return the `state` parameter to us when it redirects
+#    after a successful authorization [1].
+# 2. Canvas Studio does not allow the use of unencrypted HTTP or `localhost`
+#    as a redirect URL for an OAuth client.
+#
+# To work around these issues, we use the following authentication flow:
+#
+# 1. User clicks button in UI to initiate Canvas Studio authorization
+# 2. The button opens `/api/canvas_studio/oauth/authorize` in a popup
+# 3. In development only, the popup redirects to change the host from
+#    http://localhost:8001 to https://hypothesis.local:48001/, keeping the
+#    path the same.
```

Comment:
I need to add a note here or somewhere about how to set up `hypothesis.local` on macOS and Linux.

Reply:
Maybe also on the readme so everyone has it the first time setting LMS up.